sdk/src/test/java/com/horizen/bouncycastle/BouncyCastleTest.java

```diff
@@ -0,0 +1,132 @@
+package com.horizen.bouncycastle;
+
+import com.horizen.utils.Pair;
+import org.bouncycastle.math.ec.rfc8032.Ed25519;
+import org.bouncycastle.util.encoders.Hex;
+import org.junit.Test;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.*;
+
+public class BouncyCastleTest {
+
+    @Test
+    public void testKeiPair() {
+        byte[] seed = "seed1".getBytes();
+        Pair<byte[], byte[]> kp1 = com.horizen.utils.Ed25519.createKeyPair(seed);
+        Pair<byte[], byte[]> kp2 = com.horizen.utils.Ed25519.createKeyPair(seed);
+        byte[] pvk = kp1.getKey();
+        byte[] pvk2 = kp2.getKey();
+        assertTrue(Arrays.equals(pvk, pvk2));
+        byte[] msg = "message to sign".getBytes();
+        byte[] signature = com.horizen.utils.Ed25519.sign(pvk, msg, kp1.getValue());
+        boolean isVerified = com.horizen.utils.Ed25519.verify(signature, msg, kp1.getValue());
+        assertTrue(isVerified);
+        isVerified = com.horizen.utils.Ed25519.verify(signature, msg, kp2.getValue());
+        assertTrue(isVerified);
+    }
+
+    @Test
+    public void rfc_8032_test_1() {
+        String strPrivateKey = "9d61b19deffd5a60ba844af492ec2cc44449c5697b326919703bac031cae7f60";
+        String strPublicKey = "d75a980182b10ab7d54bfed3c964073a0ee172f3daa62325af021a68f707511a";
+        String strMsg = "";
+        String strSignature = "e5564300c360ac729086e2cc806e828a84877f1eb8e5d974d873e065224901555fb8821590a33bacc61e39701cf9b46bd25bf5f0595bbe24655141438e7a100b";
+        byte[] msg = Hex.decode(strMsg);
+        byte[] privateKey = Hex.decode(strPrivateKey);
+        byte[] publicKey = Hex.decode(strPublicKey);
+        byte[] signature = Hex.decode(strSignature);
+        byte[] signatureBC = new byte[64];
+        Ed25519.sign(privateKey, 0, publicKey, 0, (byte[]) null, msg, 0, msg.length, signatureBC, 0);
```

Comment:
instead of testing bouncycastle `org.bouncycastle.math.ec.rfc8032.Ed25519` we should only import and test `com.horizen.utils.Ed25519`. It's ggod to keep such tests, but please, adopt them to our `Ed25519` wrapper

Reply:
Done